### PR TITLE
[macOS] Extract PlatformView mutator clip updating

### PR DESCRIPTION
**Pull request by cbracken**
Extracts `updatePathClipViewsWithPaths:origin:` which updates the stack of AppKit clip views hosting the platform view.

This change is a cleanup change for readability and makes no semantic changes to how mutator views are applied. Existing unit tests in [FlutterMutatorViewTest.mm](https://github.com/flutter/engine/blob/main/shell/platform/darwin/macos/framework/Source/FlutterMutatorViewTest.mm) are expected to pass without changes.

This is refactoring prior to applying a fix for rotation handling.

Issue: https://github.com/flutter/flutter/issues/124490

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [X] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I signed the [CLA].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: shell/platform/darwin/macos/framework/Source/FlutterMutatorView.mm [hunk 1]
```diff
@@ -365,6 +365,30 @@ - (CGFloat)contentsScale {
...
+    FlutterPathClipView* pathClipView = [[FlutterPathClipView alloc] initWithFrame:self.bounds];
+    [_pathClipViews addObject:pathClipView];
+    [superView addSubview:pathClipView];
+  }
+  // Update bounds and apply clip paths.
+  for (size_t i = 0; i < _pathClipViews.count; ++i) {
+    FlutterPathClipView* pathClipView = _pathClipViews[i];
+    pathClipView.frame = self.bounds;
+    [pathClipView maskToPath:(__bridge CGPathRef)[paths objectAtIndex:i]
+                  withOrigin:self.frame.origin];
```
Comment (cbracken) [inline shell/platform/darwin/macos/framework/Source/FlutterMutatorView.mm:388]: Note that `self.frame.origin` and `finalBoundingRect.origin` are the same point. See line 385.

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

File: shell/platform/darwin/macos/framework/Source/FlutterMutatorView.mm [hunk 2]
```diff
@@ -397,37 +421,15 @@ - (void)applyFlutterLayer:(const FlutterLayer*)layer {
...
-    } else {
-      pathClipView = [[FlutterPathClipView alloc] initWithFrame:self.bounds];
-      [_pathClipViews addObject:pathClipView];
-      [lastView addSubview:pathClipView];
-    }
-    pathClipView.frame = self.bounds;
-    [pathClipView maskToPath:(__bridge CGPathRef)[paths objectAtIndex:i]
-                  withOrigin:finalBoundingRect.origin];
-    lastView = pathClipView;
-  }
```
Comment (cbracken) [inline shell/platform/darwin/macos/framework/Source/FlutterMutatorView.mm:423]: We've got two arrays here:
1. _pathClipViews (which we want to update)
2. paths (which we're updating from

Lines 401-406 adjust the length of _pathClipViews down to the length of paths if paths is longer.

Lines 410-423 update existing _pathClipViews (lines 412-413 + 419-422) or add additional ones if paths is longer than _pathClipViews (lines 414-422).

The refactor breaks this into three parts:
* Shorten _pathClipViews if needed
* Add additional _pathClipViews if needed
* Iterate over the whole set of _pathClipViews and set the frame + clip paths.

File: shell/platform/darwin/macos/framework/Source/FlutterMutatorView.mm [hunk 3]
```diff
@@ -397,37 +421,15 @@ - (void)applyFlutterLayer:(const FlutterLayer*)layer {
...
-      pathClipView = _pathClipViews[i];
-    } else {
-      pathClipView = [[FlutterPathClipView alloc] initWithFrame:self.bounds];
-      [_pathClipViews addObject:pathClipView];
-      [lastView addSubview:pathClipView];
-    }
-    pathClipView.frame = self.bounds;
-    [pathClipView maskToPath:(__bridge CGPathRef)[paths objectAtIndex:i]
-                  withOrigin:finalBoundingRect.origin];
-    lastView = pathClipView;
```
Comment (cbracken) [inline shell/platform/darwin/macos/framework/Source/FlutterMutatorView.mm:422]: `lastView` starts as `self` and as we iterate through the clip view stack, is updated to the most deeply nested clip view as we go. In the refactor, we just look up the most deeply nested clip view when we need it instead.

**Review by cyanglaz — APPROVED** [review]
LGTM

**Review by loic-sharma — APPROVED** [review]
LGTM

**Comment by Hixie** [comment]
test-exempt: code refactor with no semantic change

**Pull request merged**